### PR TITLE
Support ordered case-insensitive duration strings for timedelta

### PR DESCRIPTION
**Pull request by ddelange**
Duration strings like `7m7s` are widely used for configs, especially anything kubernetes related (anything go-related really, because of the `time.ParseDuration` builtin).

This PR adds a simple yet robust regex parser for such strings.

For timedelta specifically, this also incorporates #297

### CODE REVIEW

**Comment by sloria** [comment]
Thanks for the PR! This is interesting. Do you have a source from which you derived the regex?

I'm open to this feature, but it'd be good to have a spec or a source that we can refer to if we need to update it.

**Comment by ddelange** [comment]
Hi @sloria :wave:

Thanks for the quick reply!
- The go ParseDuration implementation has “ns”, “us” (or “µs”), “ms”, “s”, “m”, “h”.
- ISO 8601 time interval looks like P3Y6M4DT12H30M5S (P for period, and then partial overlap with go).
- Python timedelta has a subset of both, plus `weeks`.

The PR now supports all units supported by Python timedelta. I wrote the pattern myself and added the comments for maintainability.

**Comment by sloria** [comment]
Thanks, that's very helpful info 👍 . Above the regex, can you add a link to the spec and add a comment block with the discrepancies you listed?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Changelog
 
-## 11.1.0 (2025-11-11)
```
Comment (sloria) [inline CHANGELOG.md:3]: good catch

**Comment by ddelange** [comment]
@sloria 
- docs added
- had another look at the spec and added support for `0s` `0h` etc
- had another look at the marshmallow implementation and added support for `-42s` etc
- expanded tests for these cases, added some expected failures

**Comment by sloria** [comment]
Thank you! Looks great!

Is there a good reason to make the regex case-insensitive? It could be more future-proof to keep it case-sensitive. Uppercase characters are probably a typo by the user anyway, right?

**Comment by ddelange** [comment]
Sure thing, done 💥

**Comment by sloria** [comment]
Thank you! This looks good to go

**Comment by sloria** [comment]
Released in 11.2.0

**Comment by ddelange** [comment]
awesome, thanks for the collab and for quickly landing this!
landed

**Comment by ddelange** [comment]
interesting, marshmallow expects a int for timedelta, and this PR switched to subclassing TimeDelta. maybe there was some lenience that was silently swallowing the float

**Pull request merged**